packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts

```diff
@@ -169,29 +207,35 @@ export default class AmazonPayV2PaymentProcessor {
             },
         } = getPaymentMethodOrThrow(methodId);
 
-        const {
-            checkoutSettings: { features },
-            storeProfile: { shopPath, storeCountryCode },
-        } = getStoreConfigOrThrow();
-
-        const cart = getCart();
-
         if (!merchantId || !ledgerCurrency) {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const buttonBaseConfig = {
+        const buttonBaseConfig: AmazonPayV2ButtonConfig = {
             merchantId,
             ledgerCurrency,
             checkoutLanguage,
-            productType:
-                cart && getShippableItemsCount(cart) === 0
-                    ? AmazonPayV2PayOptions.PayOnly
-                    : AmazonPayV2PayOptions.PayAndShip,
             placement,
             buttonColor: AmazonPayV2ButtonColor.Gold,
+            sandbox: !!testMode,
```

Comment:
You should set this value in your if statement instead, because not all flows require it.